roles/ec2-app/tasks/nfs-templates.yml

```diff
@@ -1,30 +1,40 @@
 - block:
   - set_fact:
       nfs_host: "{{ groups['nfs-' + target][0] }}"
+
   - set_fact:
       nfs_launch_time: "{{ hostvars[nfs_host]['create_time'][0:19]|regex_replace('T|Z', ' ') }}"
+      nfs_owner: false
+      nfs_group: false
+      nfs_mode: false
 
-  - set_fact:
-      nfs_owner: "{{ appdata.nfs_template_ownership.owner }}"
-    when:
-      - appdata.nfs_template_ownership.owner is defined
+  - block:
+    - set_fact:
+        nfs_owner: "{{ appdata.nfs_template_ownership.owner }}"
+      when: "'owner' in appdata.nfs_template_ownership"
 
-  - set_fact:
-      nfs_group: "{{ appdata.nfs_template_ownership.group }}"
-    when:
-      - appdata.nfs_template_ownership.group is defined
+    - set_fact:
+        nfs_group: "{{ appdata.nfs_template_ownership.group }}"
+      when: "'group' in appdata.nfs_template_ownership"
 
-  - set_fact:
-      nfs_mode: "{{ appdata.nfs_template_ownership.mode }}"
+    - set_fact:
+        nfs_mode: "{{ appdata.nfs_template_ownership.mode }}"
+      when: "'mode' in appdata.nfs_template_ownership"
     when:
-      - appdata.nfs_template_ownership.mode is defined
+      - " appdata is defined "
+      - " 'nfs_template_ownership' in appdata "
+
+  - name: Print nfs template overide defaults for readability.
+    debug:
+      msg: " nfs template user:group:mode overide defaults of {{ nfs_owner |default('False', true) }},{{ nfs_group |default('False', true) }},{{ nfs_mode |default('False', true) }}"
 
 
+  # Ensure both times are UTC for comparison. AWS currently returns zulu time (UTC).
   - name: Test that NFS server is older than 6 minutes
-    shell: date +'%Y-%d-%m %H:%M:%S'
+    shell: date --utc +'%Y-%m-%d %H:%M:%S'
     register: nfs_ready
-    until: ((nfs_ready.stdout|to_datetime) - (nfs_launch_time|to_datetime('%Y-%m-%d %H:%M:%S'))).total_seconds()|int > nfs_wait|default(360)
-    retries: 10
+    until: ((nfs_ready.stdout|to_datetime('%Y-%m-%d %H:%M:%S')) - (nfs_launch_time|to_datetime('%Y-%m-%d %H:%M:%S'))).total_seconds()|int > nfs_wait|default(360)
+    retries: 13
     delay: 30
```

Comment:
L37: you've increased the retries only by a number of '3'.
That leads me to think this was becase you had timeouted runs that happened between 10 and 12 retries.
Consider a longer margin for failure then.
If this action is likely to fail because we're too impacient to wait a bit longer then we would be increasing the flakyness level here and and wasting time by failed ansible runs.

maybe bump it to 24 ? if you see that it takes around 12 retries